null_encoder.go

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+import (
+	"io"
+	"time"
+)
+
+// nullEncoder is an Encoder implementation that throws everything away.
+type nullEncoder struct{}
+
+// NewNullEncoder creates a fast, no-allocation encoder.
+func NewNullEncoder() Encoder {
+	return nullEncoder{}
+}
+
+func (nullEncoder) Free() {
+}
+
+// AddString throws away a string key and value.
```

Comment:
I love all these comments, but I don't think they're actually necessary since this is a private type implementing an interface which has godoc already :) You should be able to compress this down to:
```go
func (nullEncoder) AddString(_, _ string)    {}
func (nullEncoder) AddBool(_ string, _ bool) {}
func (nullEncoder) AddInt(_ string, _ int)   {}
[...]
```

(in fact, gofmt will align the `{}` for you to make the compressed version look nice)